File: app/src/main/java/com/android/orient/sports/happysports/kldf/BaseApplication.java
package com.android.orient.sports.happysports.kldf;

import android.app.Application;

import com.android.orient.sports.happysports.kldf.util.CacheUtil;

/**
 * Created by zhanglei on 2017/4/17.
 */

public class BaseApplication extends Application {

    public static Application instance;

    public static Application getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        CacheUtil.initCache(this);
    }
}

File: app/src/main/java/com/android/orient/sports/happysports/application/BaseApplication.kt
package com.android.orient.sports.happysports.application

import android.app.Application
import com.android.orient.sports.happysports.utils.ContextDelegate
import com.android.orient.sports.happysports.utils.startAlarmService

/**
 * Package Name: com.android.orient.sports.happysports.application
 * Date: 2018/11/5 16:08
 * Author: zhanglei
 */
class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        ContextDelegate.context = this
        startAlarmService(this)
    }

}